paddle/cinn/optim/elinimate_common_factor_of_local_index.cc

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) 2024 CINN Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/cinn/optim/elinimate_common_factor_of_local_index.h"
+
+#include <unordered_map>
+
+#include "paddle/cinn/common/cas.h"
+#include "paddle/cinn/ir/ir.h"
+#include "paddle/cinn/ir/ir_mutator.h"
+#include "paddle/cinn/ir/ir_printer.h"
+#include "paddle/cinn/ir/utils/ir_copy.h"
+#include "paddle/cinn/utils/string.h"
+
+namespace cinn {
+namespace optim {
+namespace {
+
+class GatherLocalIndexVisitor : public ir::IRMutator<> {
+ public:
+  void operator()(ir::Expr* expr) { ir::IRMutator<>::Visit(expr, expr); }
+
+  const std::unordered_map<std::string, std::vector<std::vector<ir::Expr>>>&
+  local_var_to_indexes() {
```

Comment:
This function doesn't change the member, should we also mark a suffix `const`?

Reply:
Done, thanks!